src/plugins/interactive_setup/server/plugin.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type { CorePreboot, PrebootPlugin, PluginInitializerContext } from 'src/core/server';
+import fs from 'fs/promises';
+import { errors } from '@elastic/elasticsearch';
+import Boom from '@hapi/boom';
+import chalk from 'chalk';
+import type { ConfigType } from './config';
+
+export class InteractiveSetupPlugin implements PrebootPlugin {
+  readonly #initializerContext: PluginInitializerContext<ConfigType>;
+  constructor(initializerContext: PluginInitializerContext<ConfigType>) {
+    this.#initializerContext = initializerContext;
+  }
+
+  public setup(core: CorePreboot) {
+    const { enabled } = this.#initializerContext.config.get<ConfigType>();
+
+    const isDev = this.#initializerContext.env.mode.dev;
+    const isConfigured =
```

Comment:
__note:__ I believe we should also treat Kibana as configured even if only `hosts` are changed - by default we have just one host, but if user changes it or adds more, or remove default - we should skip interactive setup mode. Something like this maybe:
```ts
const isElasticsearchConnectionManuallyConfigured =
      core.elasticsearch.config.credentialsSpecified ||
      core.elasticsearch.config.hosts.length !== 1 ||
      core.elasticsearch.config.hosts[0] !== 'http://localhost:9200';
```

Reply:
I forgot to mention this previously, but I _think_ even if `isElasticsearchConnectionManuallyConfigured` turns out to be `false` we still should try to call _some_ ES API (?) with the default configuration and see if we can successfully do this - and if we can we also should skip interactive setup. For example, before setting up Kibana, user could explicitly disable security in Elasticsearch and hence Kibana wouldn't need any additional configuration. This would also be the case for many existing configuration when users put TLS terminating and authenticating proxies in front of Kibana and Elasticsearch that do all the job for us.

What do you think?

Reply:
Great idea. I'll add that in.